single_cell_condensed_sdrf.sh

```diff
@@ -203,20 +203,30 @@ use_cell_types_In_condensed() {
       if [ -n "$col_num_ct" ]; then 
         awk -v fieldName="$field_to_extract" -F'\t' 'BEGIN { OFS = "\t" } NR == FNR { cell[$1]; type[$1]=$2; next } $3 in cell { print $1, $2, $3, "factor", fieldName, type[$3] }' \
             <( awk -F'\t' -v cellCol=$col_num_cell_id -v ctCol=$col_num_ct 'BEGIN { OFS = "\t" } { print $cellCol, $ctCol }' $CT ) \
-            <( awk -F'\t' 'BEGIN { OFS = "\t" } {print $1, $2, $3}' $COND | sort | uniq) >> $COND\.with_ct
+            <( awk -F'\t' 'BEGIN { OFS = "\t" } {print $1, $2, $3}' $COND | sort | uniq) > $COND\.with_ct
+
+        actual_cell_type_field=$(head -1 $CT | awk -F"\t" -v col=$col_num_ct '{print $col}')
+        if [ -s $COND\.with_ct ]; then
+          echo "Found matches for $field_to_extract to add to the condensed..."
+          echo "Annotating $actual_cell_type_field cell types"          
+          annotateCommand="annotate_celltypes_condensed_sdrf.pl -c $COND\.with_ct -o ${CONDENSED_SDRF_TSV}_celltypes -l ${CONDENSED_SDRF_TSV}_zoomalogs -e $actual_cell_type_field $exclusions"
+          eval $annotateCommand
+          
+          if [ "$?" = "0" ]; then # zooma mapping went fine
+            cat $CONDENSED_SDRF_TSV"_celltypes" >> $COND
+            # Append cell type zommification logs to main zoomification logs
+            tail -n+2 $CONDENSED_SDRF_TSV"_zoomalogs" >> $outputDir/$expId-zoomifications-log.tsv
+            rm $CONDENSED_SDRF_TSV"_zoomalogs"
+          fi
```

Comment:
Maybe we should have a warning here to be printed to the logs that zooma mapping failed, or are we directly failing if any step returns >0?

Reply:
We do have 'set -e' active at the top, so I believe this will die on errors.